tensorflow/lite/micro/examples/hello_world/evaluate_test.cc

```diff
@@ -127,12 +128,130 @@ int LoadFloatModelAndPerformInference() {
+
+  // Get the input quantization parameters
+  float input_scale = input->params.scale;
+  int input_zero_point = input->params.zero_point;
+
+  // Quantize the input from floating-point to integer
+  int8_t x_quantized = x / input_scale + input_zero_point;
+  // Place the quantized input in the model's input tensor
+  input->data.int8[0] = x_quantized;
+  // Run the model and check that it succeeds
+  TfLiteStatus invoke_status = interpreter.Invoke();
+  if (invoke_status != kTfLiteOk) {
+    MicroPrintf("Interpreter invocation failed.");
+    return kTfLiteError;
+  }
+
+  // Obtain a pointer to the output tensor.
+  TfLiteTensor* output = interpreter.output(0);
+
+  // Get the output quantization parameters
+  float output_scale = output->params.scale;
+  int output_zero_point = output->params.zero_point;
+
+  // Obtain the quantized output from model's output tensor
+  int8_t y_pred_quantized = output->data.int8[0];
+  // Dequantize the output from integer to floating-point
+  float y_pred = (y_pred_quantized - output_zero_point) * output_scale;
+
+  // Check if the output is within a small range of the expected output
+  float epsilon = 0.05f;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
+
+  // Run inference on several more values and confirm the expected outputs
+  x = 1.f;
+  y_true = sin(x);
+  input->data.int8[0] = x / input_scale + input_zero_point;
+  interpreter.Invoke();
+  y_pred = (output->data.int8[0] - output_zero_point) * output_scale;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
+
+  x = 3.f;
+  y_true = sin(x);
+  input->data.int8[0] = x / input_scale + input_zero_point;
+  interpreter.Invoke();
+  y_pred = (output->data.int8[0] - output_zero_point) * output_scale;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
+
+  x = 5.f;
+  y_true = sin(x);
+  input->data.int8[0] = x / input_scale + input_zero_point;
+  interpreter.Invoke();
+  y_pred = (output->data.int8[0] - output_zero_point) * output_scale;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
+
+  return kTfLiteOk;
+}
+
 int main(int argc, char* argv[]) {
   int status = LoadFloatModelAndPerformInference();
+  int status_1 = LoadQuantModelAndPerformInference();
   // To be part of the unit test suite, each test file needs to print out
   // either one of the following strings. These strings are required to
   // be considered as a unit test for the tflm makefiles.
-  if (status == kTfLiteOk) {
+  if (status == kTfLiteOk && status_1 == kTfLiteOk) {
```

Comment:
this is quite error prone and hard to read.

Let's use TF_LITE_ENSURE_STATUS instead.

something like:
```cc
TF_LITE_ENSURE_STATUS(LoadFloatModelAndPerformInference());
TF_LITE_ENSURE_STATUS(LoadQuantModelAndPerformInference());

// If all tests pass, return specific string for the unit testing framework.
MicroPrintf("~~~ALL TESTS PASSED~~\n");
return kTfLiteOk;
```

I don't think we need to have a `MicroPrintf("~~~SOME TESTS FAILED~~~\n");` in case of a failure. It would be good to confirm that that is the case by manually introducing an error.

Reply:
I tested it with a failed test and the failure is surfacing without the Log. So, I am removing all logs from main.

Reply:
After I pushed the tests without the log - hexagon, bluepill and stm32f4 builds are failing. So added the log back.